### ISSUE TITLE
Feature causes non-convergence

### ISSUE THREAD

**Issue by systats** [rstudio/keras#452]
Hi folks,

I want to predict the outcome of a esports game (LOL) but one feature **Gold at minute 15**  causes the model not to converge. Interestingly, the same set of features works with other machine learning libraries (h2o) very well. 

The data looks as follows:

```
library(keras)
library(dplyr)
dat <- get(load(url("https://www.dropbox.com/s/boycb7h8pqvpn83/example_data.Rdata?dl=1")))
glimpse(dat)
# Observations: 6887 
# Variables: 7
# $ blue_win       <dbl> 1, 1, 1, 1, 1
# $ blue_gold_15   <dbl> 23.641, 24.707, 22.736, 24.531, 23.130
# $ red_gold_15    <dbl> 22.592, 24.945, 22.658, 21.969, 22.894
# $ blue_kills_15  <int> 3, 3, 0, 3, 1
# $ red_kills_15   <int> 1, 3, 0, 1, 2
# $ blue_towers_15 <int> 1, 1, 1, 2, 1
# $ red_towers_15  <int> 0, 5, 1, 0, 0
```

A minimal example of a very simple model:

```
fit1 <- keras_model_sequential() %>% 
  layer_dense(units=1, activation = 'sigmoid', input_shape = c(6)) %>% 
  compile(
    loss = "binary_crossentropy",
    optimizer = "adam",
    metrics = "accuracy"
  )

summary(fit1)

fit1 %>% 
  keras::fit(
    x = as.matrix(dat[,-1]), 
    y = matrix(dat[,1], ncol = 1),
    batch_size = 3,
    epochs = 2, 
    validation_split = .2
  )

# Train on 6887 samples, validate on 1722 samples
# Epoch 1/2
# 6887/6887 [==============================] - 4s 598us/step - loss: 6.2859 - acc: 0.0939 - val_loss: 7.4805 - val_acc: 0.0000e+00
# Epoch 2/2
# 6887/6887 [==============================] - 3s 490us/step - loss: 7.2455 - acc: 0.0000e+00 - val_loss: 7.4805 - val_acc: 0.0000e+00
```

What I tried so far:

* not included gold: works perfectly >70% ACC
* different transformations for gold (range01, scale): no difference
* other libraries learn very well from gold

As I am running out of ideas it would be great if someone could give me advice. Thanks in advance!

**Comment by skeydan** [comment]
Hi,

the respective column(s) have missing values. If you do

```
dat <- na.omit(dat)
```

it will run and reach ~ 0.72 validation accuracy after 30 epochs.

(Maybe h2o removes missing values itself, but Keras doesn't.)

**Comment by systats** [comment]
Well that was simple :D Thanks for the tip **skeydan**!
